expression/builtin_cast_vec.go

```diff
@@ -530,11 +530,32 @@ func (b *builtinCastRealAsTimeSig) vecEvalTime(input *chunk.Chunk, result *chunk
 }
 
 func (b *builtinCastDecimalAsDecimalSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCastDecimalAsDecimalSig) vecEvalDecimal(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	if err := b.args[0].VecEvalDecimal(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	n := input.NumRows()
+	decs := result.Decimals()
+	sc := b.ctx.GetSessionVars().StmtCtx
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		dec := &types.MyDecimal{}
+		if !(b.inUnion && mysql.HasUnsignedFlag(b.tp.Flag) && decs[i].IsNegative()) {
```

Comment:
We can cache the result of `b.inUnion && mysql.HasUnsignedFlag(b.tp.Flag)` outside the for loop.

Reply:
> We can cache the result of `b.inUnion && mysql.HasUnsignedFlag(b.tp.Flag)` outside the for loop.

Done.